### PR TITLE
impl: add integration tests for LRO Start and Await methods

### PR DESCRIPTION
**Pull request by scotthart**
part of the work for #7658

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/14377)

### CODE REVIEW

File: google/cloud/spanner/admin/integration_tests/instance_admin_integration_test.cc [hunk 1]
```diff
@@ -214,6 +214,44 @@ TEST_F(InstanceAdminClientTest, InstanceCRUDOperations) {
...
+  if (!Emulator() && !RunSlowInstanceTests()) {
+    GTEST_SKIP() << "skipping slow instance tests; set "
+                 << "GOOGLE_CLOUD_CPP_SPANNER_SLOW_INTEGRATION_TESTS=instance"
+                 << " to override";
+  }
+
+  std::string instance_id = spanner_testing::RandomInstanceName(generator_);
+  Instance in(ProjectId(), instance_id);
+  ASSERT_FALSE(in.project_id().empty());
+  ASSERT_FALSE(in.instance_id().empty());
```
Comment (dbolduc) [inline google/cloud/spanner/admin/integration_tests/instance_admin_integration_test.cc:227]: nit: remove?

I see that the above test does the same thing, but `RandomInstanceName` is never going to return an empty instance the way `ProjectId()` or `InstanceId()` might.

File: google/cloud/spanner/admin/integration_tests/instance_admin_integration_test.cc [hunk 2]
```diff
@@ -214,6 +214,44 @@ TEST_F(InstanceAdminClientTest, InstanceCRUDOperations) {
...
+      client_.CreateInstance(ExperimentalTag{}, NoAwaitTag{},
+                             CreateInstanceRequestBuilder(in, config_name)
+                                 .SetDisplayName("test-display-name")
+                                 .SetNodeCount(1)
+                                 .SetLabels({{"label-key", "label-value"}})
+                                 .Build());
+  ASSERT_STATUS_OK(operation);
+  auto instance_config =
+      client_.CreateInstanceConfig(ExperimentalTag{}, *operation).get();
+  EXPECT_THAT(instance_config, StatusIs(StatusCode::kInvalidArgument));
```
Comment (dbolduc) [inline google/cloud/spanner/admin/integration_tests/instance_admin_integration_test.cc:246]: This confused me initially. Maybe a comment like:

```cc
// Verify that the client rejects LROs with the wrong metadata type.
```

File: google/cloud/spanner/admin/integration_tests/instance_admin_integration_test.cc [hunk 3]
```diff
@@ -214,6 +214,44 @@ TEST_F(InstanceAdminClientTest, InstanceCRUDOperations) {
...
+
+  auto instance = client_.CreateInstance(ExperimentalTag{}, *operation).get();
+  ASSERT_STATUS_OK(instance);
+  EXPECT_THAT(instance->name(), Eq(in.FullName()));
+  EXPECT_EQ(instance->display_name(), "test-display-name");
```
Comment (dbolduc) [inline google/cloud/spanner/admin/integration_tests/instance_admin_integration_test.cc:251]: nit: local consistency of these?

This file seems to prefer `EXPECT_EQ`

File: google/cloud/spanner/admin/integration_tests/instance_admin_rest_integration_test.cc [hunk 4]
```diff
@@ -257,6 +258,44 @@ TEST_F(InstanceAdminClientRestTest, InstanceCRUDOperations) {
...
+                                 .Build());
+  ASSERT_STATUS_OK(operation);
+  auto instance_config =
+      client_.CreateInstanceConfig(ExperimentalTag{}, *operation).get();
+  EXPECT_THAT(instance_config, StatusIs(StatusCode::kInvalidArgument));
+
+  auto instance = client_.CreateInstance(ExperimentalTag{}, *operation).get();
+  ASSERT_STATUS_OK(instance);
+  EXPECT_THAT(instance->name(), Eq(in.FullName()));
+  EXPECT_EQ(instance->display_name(), "test-display-name");
```
Comment (dbolduc) [inline google/cloud/spanner/admin/integration_tests/instance_admin_rest_integration_test.cc:295]: same

File: google/cloud/spanner/admin/integration_tests/instance_admin_rest_integration_test.cc [hunk 5]
```diff
@@ -257,6 +258,44 @@ TEST_F(InstanceAdminClientRestTest, InstanceCRUDOperations) {
...
+  if (!Emulator() && !RunSlowInstanceTests()) {
+    GTEST_SKIP() << "skipping slow instance tests; set "
+                 << "GOOGLE_CLOUD_CPP_SPANNER_SLOW_INTEGRATION_TESTS=instance"
+                 << " to override";
+  }
+
+  std::string instance_id = spanner_testing::RandomInstanceName(generator_);
+  Instance in(ProjectId(), instance_id);
+  ASSERT_FALSE(in.project_id().empty());
+  ASSERT_FALSE(in.instance_id().empty());
```
Comment (dbolduc) [inline google/cloud/spanner/admin/integration_tests/instance_admin_rest_integration_test.cc:271]: same nit about deleting.

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/14377)* status: 0 of 3 files reviewed, 7 unresolved discussions (waiting on @dbolduc)

___
*[`google/cloud/compute/integration_tests/compute_integration_test.cc` line 94 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/14377#gh-1655549245:-O0PFvdOAyNm6v0jPSt3:bx6q95p) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/7a566338104b1a4f19b16221b08086269ac7eded/google/cloud/compute/integration_tests/compute_integration_test.cc#L94)):*
<details><summary><i>Previously, dbolduc (Darren Bolduc) wrote…</i></summary><blockquote>

nit (predates this PR): `using ::google::cloud::testing_util::IsOk()`
</blockquote></details>

Fixed throughout the file.
___
*[`google/cloud/compute/integration_tests/compute_integration_test.cc` line 102 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/14377#gh-1655571026:-O0PFxar9xyIwkrqKuog:b-2gwiq2) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/7a566338104b1a4f19b16221b08086269ac7eded/google/cloud/compute/integration_tests/compute_integration_test.cc#L102)):*
<details><summary><i>Previously, dbolduc (Darren Bolduc) wrote…</i></summary><blockquote>

Why the string serializing/parsing?

```cc
  auto operation = client.InsertDisk(ExperimentalTag{}, NoAwaitTag{},
                                        project_id_, zone_, disk);
  ASSERT_THAT(operation, IsOk());
  auto result = client.InsertDisk(ExperimentalTag{}, *operation).get();
```
</blockquote></details>

Added comment to explain.
___
*[`google/cloud/spanner/admin/integration_tests/instance_admin_integration_test.cc` line 227 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/14377#gh-1655588862:-O0PFzohDgBX7eae5E2H:ba3dfob) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/7a566338104b1a4f19b16221b08086269ac7eded/google/cloud/spanner/admin/integration_tests/instance_admin_integration_test.cc#L227)):*
<details><summary><i>Previously, dbolduc (Darren Bolduc) wrote…</i></summary><blockquote>

nit: remove?

I see that the above test does the same thing, but `RandomInstanceName` is never going to return an empty instance the way `ProjectId()` or `InstanceId()` might.
</blockquote></details>

Removed
___
*[`google/cloud/spanner/admin/integration_tests/instance_admin_integration_test.cc` line 246 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/14377#gh-1655594472:-O0PG-pp9_fT-FzE1ECy:bs3mhdg) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/7a566338104b1a4f19b16221b08086269ac7eded/google/cloud/spanner/admin/integration_tests/instance_admin_integration_test.cc#L246)):*
<details><summary><i>Previously, dbolduc (Darren Bolduc) wrote…</i></summary><blockquote>

This confused me initially. Maybe a comment like:

```cc
// Verify that the client rejects LROs with the wrong metadata type.
```
</blockquote></details>

Comment added.
___
*[`google/cloud/spanner/admin/integration_tests/instance_admin_integration_test.cc` line 251 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/14377#gh-1655597081:-O0PG0uv4nOCCDUV2pCs:b-886gsx) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/7a566338104b1a4f19b16221b08086269ac7eded/google/cloud/spanner/admin/integration_tests/instance_admin_integration_test.cc#L251)):*
<details><summary><i>Previously, dbolduc (Darren Bolduc) wrote…</i></summary><blockquote>

nit: local consistency of these?

This file seems to prefer `EXPECT_EQ`
</blockquote></details>

Fixed
___
*[`google/cloud/spanner/admin/integration_tests/instance_admin_rest_integration_test.cc` line 271 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/14377#gh-1655597676:-O0PGYQMEC_ffffus2xc:bss4a2b) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/7a566338104b1a4f19b16221b08086269ac7eded/google/cloud/spanner/admin/integration_tests/instance_admin_rest_integration_test.cc#L271)):*
<details><summary><i>Previously, dbolduc (Darren Bolduc) wrote…</i></summary><blockquote>

same nit about deleting.
</blockquote></details>

Removed.
___
*[`google/cloud/spanner/admin/integration_tests/instance_admin_rest_integration_test.cc` line 295 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/14377#gh-1655597519:-O0PGZG17eep0uNLNFAp:b-886gsx) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/7a566338104b1a4f19b16221b08086269ac7eded/google/cloud/spanner/admin/integration_tests/instance_admin_rest_integration_test.cc#L295)):*
<details><summary><i>Previously, dbolduc (Darren Bolduc) wrote…</i></summary><blockquote>

same
</blockquote></details>

Fixed

**Review by dbolduc — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/14377)* status: 0 of 3 files reviewed, 7 unresolved discussions (waiting on @scotthart)

___
*[`google/cloud/compute/integration_tests/compute_integration_test.cc` line 102 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/14377#gh-1655571026:-O0PXMm5DVLyn3oP2cXK:bmbogru) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/7a566338104b1a4f19b16221b08086269ac7eded/google/cloud/compute/integration_tests/compute_integration_test.cc#L102)):*
<details><summary><i>Previously, scotthart (Scott Hart) wrote…</i></summary><blockquote>

Added comment to explain.
</blockquote></details>

Ah. It feels like we are testing protobuf, but that is fine.

**Pull request merged**